### PR TITLE
feat: add wrap with observer assist

### PR DESCRIPTION
**Pull request by amondnet**
Describe the changes proposed in this Pull Request.

If the PR fixes a specific issue, reference the issue with **`Fixes #`**.

wrap_with_observer

---
## Pull Request Checklist

- [x] If the changes are being made to code, ensure the **version in `pubspec.yaml`** is updated. 
- [x] Increment the **`major`/`minor`/`patch`/`patch-count`**, depending on the complexity of change
- [x] Add the necessary **unit tests** to ensure the coverage does not drop
- [x] Update the **Changelog** to include all changes made in this PR, organized by version
- [x] Run the **`melos run set_version` command** from the root directory
- [x] Include the **necessary reviewers** for the PR
- [x] Update the **docs** if there are any API changes or additions to functionality

### CODE REVIEW

**Review by pavanpodila — APPROVED** [review]

File: mobx_lint/pubspec.yaml [hunk 1]
```diff
@@ -0,0 +1,21 @@
+name: mobx_lint
+description: A starting point for Dart libraries or applications.
+version: 1.0.0
+homepage: https://mobx.dev
```
Comment (pavanpodila) [inline mobx_lint/pubspec.yaml:4]: Need to update these properties with proper descriptions.

**Review by pavanpodila — CHANGES_REQUESTED** [review]

**Comment by pavanpodila** [comment]
Looks like a cool addition. We could do a lot more with this I think. A few ideas:

1. Flag Observers that don't have an observable in the builder?
2. Add reaction in the initState of a StatefulWidget with proper dispose
3. Generate stubs for wrapping an observable mutation to happen inside a `runInAction()`

File: mobx_lint/pubspec.yaml (hunk 1 above)
Comment (amondnet) [reply mobx_lint/pubspec.yaml]: @pavanpodila done.

**Comment by subzero911** [comment]
image
Already exists in `flutter_mobx` VsCode extension.
But! it fails on some very large pieces of code.

File: mobx_lint/pubspec.yaml [hunk 2]
```diff
@@ -0,0 +1,21 @@
+name: mobx_lint
```
Comment (pavanpodila) [inline mobx_lint/pubspec.yaml:1]: for publishing this package, we need to make changes to the github workflow as well...pls see

**Review by pavanpodila — CHANGES_REQUESTED** [review]

File: mobx_lint/pubspec.yaml (hunk 2 above)
Comment (amondnet) [reply mobx_lint/pubspec.yaml:1]: @pavanpodila done

File: mobx_lint/README.md [hunk 3]
```diff
@@ -0,0 +1,52 @@
+mobx_lint is a developer tool for users of mobx, designed to help stop common issues and simplify repetitive tasks.
```
Comment (pavanpodila) [inline mobx_lint/README.md:1]: I think we should mention this in the docs as well so folks can see it easily on the website ... perhaps it can be a separate PR

**Review by pavanpodila — APPROVED** [review]

**Comment by pavanpodila** [comment]
Hi @amondnet, can we move this package under the mobx publisher ?
